### PR TITLE
Checkcoords

### PR DESCRIPTION
**Pull request by roxyboy**
Edits to make sure the frequency spacing is always positive even with descending coordinate input.

### CODE REVIEW

**Comment by roxyboy** [comment]
Not sure why Travis isn't working....?

**Comment by rabernat** [comment]
Travis has recently broken on several of my repositories, so I would be very glad if you could figure this out. Travis did [run](https://travis-ci.org/github/xgcm/xrft/builds/676329581), it just didn't post here.

File: xrft/xrft.py [hunk 1]
```diff
@@ -245,7 +245,7 @@ def _new_dims_and_coords(da, axis_num, dim, wavenm, prefix):
 
     dk = [l[1] - l[0] for l in wavenm]
     for this_dk, d in zip(dk, dim):
-        newcoords[prefix + d + '_spacing'] = this_dk
+        newcoords[prefix + d + '_spacing'] = np.abs(this_dk)
```
Comment (rabernat) [inline xrft/xrft.py:248]: I would slightly prefer to fix this a little farther upstream, where we calculate dx.

**Comment by roxyboy** [comment]
Travis seems to be back posting here now...?

File: xrft/xrft.py [hunk 2]
```diff
@@ -346,7 +346,7 @@ def dft(da, spacing_tol=1e-3, dim=None, real=None, shift=True, detrend=None,
         if pd.api.types.is_timedelta64_dtype(diff):
             # convert to seconds so we get hertz
             diff = diff.astype('timedelta64[s]').astype('f8')
-        delta = diff[0]
+        delta = np.abs(diff[0])
```
Comment (roxyboy) [inline xrft/xrft.py:349]: @rabernat I moved the fix to where we calculate dx, as you suggested :)

**Comment by roxyboy** [comment]
Could we merge this @rabernat ...?

**Comment by rabernat** [comment]
> Could we merge this @rabernat ...?

I *think* I am okay with this if you can address my main comment above (mistake in test). If you want to skip fixing all the whitespace, that's fine, but please do try to keep this in mind for the future.

File: xrft/tests/test_xrft.py [hunk 3]
```diff
@@ -183,6 +183,12 @@ def test_dft_2d(self):
         da_prime = (da - da.mean(dim=dim)).values
         npt.assert_almost_equal(ft.values, np.fft.fftn(da_prime*window))
 
+        da = xr.DataArray(np.random.rand(N,N), dims=['x','y'],
+                         coords={'x':range(N,0,-1),'y':range(N,0,-1)}
+                         )
+        assert xrft.power_spectrum(da, shift=False,
+                                  density=True).data.all() >= 0.
```
[the remark this answers is not in the document]
Comment (roxyboy) [reply xrft/tests/test_xrft.py:190]: That's one thing weird I noticed also that these whitespace changes are showing up although I never touched those spaces...

**Comment by roxyboy** [comment]
Ok, I will fix the test. I really honestly don't know why these whitespace changes occurred (I really didn't touch anything besides the section where I change the frequency spacings to absolute...) but I will try to be careful...

**Comment by rabernat** [comment]
It may be something that your editor does automatically every time you save a file.

**Pull request merged**